presto-hive/src/test/java/com/facebook/presto/hive/s3/TestPrestoS3FileSystem.java

```diff
@@ -793,6 +800,109 @@ public ObjectListing listObjects(ListObjectsRequest listObjectsRequest)
         }
     }
 
+    @Test
+    public void testMultipart()
```

Comment:
Can you also add a test that writes a complete pre-canned file using the PrestoS3FileSystem and reads it back and asserts on its contents ? Testing around boundary cases (under min part size, just above min part size, large file size) will prove robustness

Reply:
Added here: https://github.com/prestodb/presto/blob/ec82b5ea7861e87960288fae99956e3ffde98787/presto-hive/src/test/java/com/facebook/presto/hive/s3/TestPrestoS3FileSystem.java#L910